src/ui/DynamicHierarchicalFacetSearch/DynamicHierarchicalFacetSearchValueRenderer.ts

```diff
@@ -0,0 +1,118 @@
+import { IDynamicHierarchicalFacet } from '../DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { DynamicHierarchicalFacetSearchValue } from './DynamicHierarchicalFacetSearchValue';
+import { $$ } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+
+const PATH_START_LENGTH = 1;
+const PATH_END_LENGTH = 2;
+const PATH_SEPARATOR = '/';
+
+const VALUE_CLASSNAME = 'coveo-dynamic-hierarchical-facet-search-value';
+const HEADER_CLASSNAME = `${VALUE_CLASSNAME}-header`;
+const LABEL_CLASSNAME = `${VALUE_CLASSNAME}-label`;
+const COUNT_CLASSNAME = `${VALUE_CLASSNAME}-results-count`;
+const PATH_CLASSNAME = `${VALUE_CLASSNAME}-path`;
+const PATH_ELLIPSIS_CLASSNAME = `${PATH_CLASSNAME}-ellipsis`;
+const PATH_PART_CLASSNAME = `${PATH_CLASSNAME}-part`;
+const PATH_SEPARATOR_CLASSNAME = `${PATH_CLASSNAME}-separator`;
+
+export const DynamicHierarchicalFacetSearchValueRendererClassNames = {
+  VALUE_CLASSNAME,
+  HEADER_CLASSNAME,
+  LABEL_CLASSNAME,
+  COUNT_CLASSNAME,
+  PATH_CLASSNAME,
+  PATH_ELLIPSIS_CLASSNAME,
+  PATH_PART_CLASSNAME,
+  PATH_SEPARATOR_CLASSNAME
+};
+
+export class DynamicHierarchicalFacetSearchValueRenderer {
+  constructor(private facetValue: DynamicHierarchicalFacetSearchValue, private facet: IDynamicHierarchicalFacet) {}
+
+  private get pathToRender(): { start: string[]; end?: string[] } {
+    const parentPath = this.facetValue.fullPath.slice(0, -1);
+    if (!parentPath.length) {
+      return { start: [this.facet.options.clearLabel] };
+    }
+    if (parentPath.length > PATH_START_LENGTH + PATH_END_LENGTH) {
+      return { start: parentPath.slice(0, PATH_START_LENGTH), end: parentPath.slice(-PATH_END_LENGTH) };
+    }
+    return { start: parentPath };
+  }
+
+  private get label() {
+    const { start, end } = this.pathToRender;
+    return l(
+      'HierarchicalFacetValueIndentedUnder',
+      l(
+        'SelectValueWithResultCount',
+        this.facetValue.displayValue,
+        l('ResultCount', this.facetValue.numberOfResults, this.facetValue.numberOfResults)
+      ),
+      [...start, ...(end || [])].reverse().join(', ')
```

Comment:
The label spells out categories in the reverse order (small to large). Would it be better the other way around?

Reply:
So a screen reader would read the labels in reverse order versus what is actually displayed ? I think that's kind of weird 🤔 ...

Reply:
Alright, I'll change it to be in the same (large to small) order.